plugins/search-backend/src/actions/createQueryAction.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ActionsRegistryService } from '@backstage/backend-plugin-api/alpha';
+import { SearchEngine } from '@backstage/plugin-search-backend-node';
+
+export const createQueryAction = ({
+  engine,
+  actionsRegistry,
+}: {
+  engine: SearchEngine;
+  actionsRegistry: ActionsRegistryService;
+}) => {
+  actionsRegistry.register({
+    name: 'query',
+    title: 'Query Search Engine',
+    description: `
+This allows you to query the search engine for documents.
+Types and filters can be used to narrow down the search results.
+Available types can be queried using the \`get-document-types\` action.
+Pagination is supported via the \`pageLimit\` and \`pageCursor\` parameters.
+Results are returned in a paginated format, along with cursors for navigating to the next and previous pages of results.
+    `,
+    attributes: {
+      readOnly: true,
+    },
+    schema: {
+      input: z =>
+        z.object({
+          term: z.string().describe('The search term to query for'),
+          types: z
+            .array(z.string())
```

Comment:
It would be nice to have support for defining the `input` in an async function, get the allowed types on demand, and add them as an `enum` here. This would render the `getDocumentTypesAction` obsolete and improve the validation.